op_crates/fetch/lib.rs

```diff
@@ -87,10 +96,10 @@ pub fn get_declaration() -> PathBuf {
   PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("lib.deno_fetch.d.ts")
 }
 
-pub async fn op_fetch<FP>(
-  state: Rc<RefCell<OpState>>,
+pub fn op_fetch<FP>(
```

Comment:
The split between `op_fetch` and `op_fetch_send` and the introduction of `FetchRequestResource` is unfortunate but is needed because the resource ID for the request body needs to be passed to the client synchronously _before_ awaiting the response. (ie JS needs to know where to stream the request body, _before_ getting back the response). I think the overhead in most cases will be negligible.